src/components/Layout/index.tsx

```diff
@@ -1,25 +1,87 @@
-import { FC, ReactNode } from 'react';
-import Link from 'next/link';
-import { useRouter } from 'next/router';
-import Footer from '@/components/footer';
-import styles from '@/components/Layout/Layout.module.scss';
-import NavBar from '@/components/navBar';
-
+import { FC, ReactNode, useEffect, useState } from "react";
+import Link from "next/link";
+import { useRouter } from "next/router";
+import Footer from "@/components/footer";
+import styles from "@/components/Layout/Layout.module.scss";
+import NavBar from "@/components/navBar";
+import Image from "next/image";
+import { toast, ToastTypes } from "@/helperFunctions/toast";
+import fetch from "@/helperFunctions/fetch";
+import { useDebounce } from "../../hooks/useDebounce";
 interface Props {
-  children?: ReactNode
+  children?: ReactNode;
+  editSetter: any;
```

Comment:
What is the purpose of editSetter here and why is the type of the same any?

Reply:
I wasn't able to figure out what its type should be so I set it to any

Reply:
What are we doing here?
We are setting a state action with a boolean value right?
Could you look more into that please